src/Config/ObjectModel/DataSource.cs

```diff
@@ -12,40 +12,62 @@ namespace Azure.DataApiBuilder.Config.ObjectModel;
 /// </summary>
 /// <param name="DatabaseType">Type of database to use.</param>
 /// <param name="ConnectionString">Connection string to access the database.</param>
-/// <param name="Options">Custom options for the specific database. If there are no options this will be empty.</param>
-public record DataSource(DatabaseType DatabaseType, string ConnectionString, Dictionary<string, JsonElement> Options)
+/// <param name="Options">Custom options for the specific database. If there are no options, this could be null.</param>
+public record DataSource(DatabaseType DatabaseType, string ConnectionString, Dictionary<string, JsonElement>? Options)
 {
     /// <summary>
     /// Converts the <c>Options</c> dictionary into a typed options object.
+    /// May return null if the dictionary is null.
     /// </summary>
     /// <typeparam name="TOptionType">The strongly typed object for Options.</typeparam>
     /// <returns>The strongly typed representation of Options.</returns>
     /// <exception cref="NotSupportedException">Thrown when the provided <c>TOptionType</c> is not supported for parsing.</exception>
-    public TOptionType GetTypedOptions<TOptionType>() where TOptionType : IDataSourceOptions
+    public TOptionType? GetTypedOptions<TOptionType>() where TOptionType : IDataSourceOptions
```

Comment:
Rather than returning `null`, should we return the default for the requested options?

Reply:
I am actually returning the default already. But since that could be null, VS complains if I dont make the return type nullable

Reply:
I meant constructing an object with default values, not returning `default`, which will return `null` since the options here are reference types not value types (if we wanted them to be value types they'd need to be `record struct` instead of `record`).

Reply:
We dont know what the default values for `CosmosDbNoSQLDataSourceOptions` should be, those are mandatory options. But if they are not specified, we need to throw an exception. To identify if they were specified or not by the customer, null check is better than trying to check for default values that we might assign to the object. 
For CosmosDbOptions, it will also be harder to distinguish between default values assigned by us v/s what the customer is providing.

I let MsSqlOptions to have default values if not specified.